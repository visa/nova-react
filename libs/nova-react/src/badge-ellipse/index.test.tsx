/**
 *              Copyright (c) 2025 Visa, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import BadgeEllipse from './';

describe('BadgeEllipse', () => {
  it('should render defaults correctly', async () => {
    const { container } = render(<BadgeEllipse />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
    expect(container.firstElementChild?.getAttribute('class')).toBe('v-icon v-icon-tiny v-ellipse');
    expect(container.firstElementChild?.tagName).toBe('svg');
  });

  it('should allow custom classNames', () => {
    const { container } = render(<BadgeEllipse className="test-class" />);
    expect(container.firstElementChild?.getAttribute('class')).toBe('v-icon v-icon-tiny v-ellipse test-class');
  });

  it('should allow custom aria-labels', () => {
    const { container } = render(<BadgeEllipse aria-label="test label" />);
    expect(container.firstElementChild?.getAttribute('aria-label')).toBe('test label');
  });

  it('should have aria-hidden if custom aria-label is not provided', () => {
    const { container } = render(<BadgeEllipse />);
    expect(container.firstElementChild?.getAttribute('aria-hidden')).toBe('true');
  });
});
