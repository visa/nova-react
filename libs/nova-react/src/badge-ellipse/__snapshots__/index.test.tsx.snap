// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeEllipse should render defaults correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="v-icon v-icon-tiny v-ellipse"
    height="16"
    style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="8"
      cy="8"
      r="8"
      style="fill: var(--v-badge-ellipse-color);"
    />
  </svg>
</div>
`;
